modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/fieldset/FieldSetDDMFormFieldTemplateContextContributor.java

```diff
@@ -63,87 +69,106 @@
 			(Map<String, List<Object>>)ddmFormFieldRenderingContext.getProperty(
 				"nestedFields");
 
+		List<Object> nestedFields = new ArrayList<>();
+
 		if (nestedFieldsMap == null) {
-			nestedFieldsMap = new HashMap<>();
+			nestedFields.addAll(
+				getNestedFields(
+					(List<Object>)ddmFormFieldRenderingContext.getProperty(
+						"fields")));
+		}
+		else {
+			nestedFields = getNestedFields(
+				nestedFieldsMap,
+				getNestedFieldNames(
+					GetterUtil.getString(
+						ddmFormField.getProperty("nestedFieldNames")),
+					nestedFieldsMap.keySet()));
 		}
 
-		List<Object> nestedFields = getNestedFields(
-			nestedFieldsMap,
-			getNestedFieldNames(
-				GetterUtil.getString(
-					ddmFormField.getProperty("nestedFieldNames")),
-				nestedFieldsMap.keySet()));
+		JSONArray rows = getJSONArray(
+			GetterUtil.getString(ddmFormField.getProperty("rows")));
+
+		if (rows.length() == 0) {
```

Comment:
Aqui a gente precisa dar prioridade ao ddmStructureId. Se tiver pega o rows de lá. Se for zero pega da propriedade rows.